backend/src/test/java/de/mightypc/backend/service/hardware/SsdServiceTest.java

```diff
@@ -145,4 +151,42 @@ void deleteById_ThrowsHardwareNotFoundExceptionWhenSsdDoesNotExist() {
         assertThrows(HardwareNotFoundException.class, () -> ssdService.deleteById(id), "Expected HardwareNotFoundException when SSD does not exist");
         verify(ssdRepository).existsById(id);
     }
+
+    @Test
+    void attachPhoto_WhenSsdExists_ThenPhotoIsAttached() {
+        // Arrange
+        String ssdId = "1";
+        String photoUrl = "https://example.com/photo.jpg";
+        SSD ssdBeforeUpdate = new SSD("1", new HardwareSpec("test", "test", new BigDecimal("1"), 1.01f), 5, 950, new ArrayList<>());
+        Optional<SSD> optionalSsdBeforeUpdate = Optional.of(ssdBeforeUpdate);
+
+        List<String> photosWithNewUrl = new ArrayList<>();
+        photosWithNewUrl.add(photoUrl);
+        SSD ssdAfterUpdate = new SSD("1", new HardwareSpec("test", "test", new BigDecimal("1"), 1.01f), 5, 950, photosWithNewUrl);
+
+        when(ssdRepository.findById(ssdId)).thenReturn(optionalSsdBeforeUpdate);
+        when(ssdRepository.save(any(SSD.class))).thenReturn(ssdAfterUpdate);
+
+        // Act
+        ssdService.attachPhoto(ssdId, photoUrl);
+
+        // Assert
+        verify(ssdRepository).findById(ssdId);
+        verify(ssdRepository).save(ssdAfterUpdate);
+    }
+
+
+    @Test
+    void attachPhoto_WhenSsdDoesNotExist_ThenNoActionTaken() {
```

Comment:
Man könnte in allen Testnamen auch noch das Wort "Test" reinschreiben, also z.B. "attachPhotoTest_WhenSsdDoesNotExist_ThenNoActionTaken()"